clustered/integration-test/src/test/java/org/ehcache/clustered/writebehind/BasicClusteredWriteBehindWithPassiveTest.java

```diff
@@ -0,0 +1,218 @@
+import org.ehcache.config.builders.ResourcePoolsBuilder;
+import org.ehcache.config.builders.WriteBehindConfigurationBuilder;
+import org.ehcache.config.units.EntryUnit;
+import org.ehcache.config.units.MemoryUnit;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.terracotta.testing.rules.Cluster;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+
+import static org.ehcache.clustered.client.config.builders.ClusteringServiceConfigurationBuilder.cluster;
+import static org.ehcache.config.builders.CacheConfigurationBuilder.newCacheConfigurationBuilder;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.terracotta.testing.rules.BasicExternalClusterBuilder.newCluster;
+
+public class BasicClusteredWriteBehindWithPassiveTest extends ClusteredTests {
+
+  private static final String RESOURCE_CONFIG =
+      "<config xmlns:ohr='http://www.terracotta.org/config/offheap-resource'>"
+      + "<ohr:offheap-resources>"
+      + "<ohr:resource name=\"primary-server-resource\" unit=\"MB\">64</ohr:resource>"
+      + "</ohr:offheap-resources>" +
+      "</config>\n";
+
+  @ClassRule
+  public static Cluster CLUSTER =
+      newCluster(2).in(new File("build/cluster")).withServiceFragment(RESOURCE_CONFIG).build();
+
+  @BeforeClass
+  public static void waitForActive() throws Exception {
+    CLUSTER.getClusterControl().waitForActive();
+    CLUSTER.getClusterControl().waitForRunningPassivesInStandby();
+  }
+
+  private static final String CACHE_NAME = "cache-1";
+  private static final long KEY = 1L;
+
+  private RecordingLoaderWriter<Long, String> loaderWriter;
+
+  @Before
+  public void setUp() {
+    loaderWriter = new RecordingLoaderWriter<>();
+  }
+
+  @Test
+  public void testBasicClusteredWriteBehind() throws Exception {
+    PersistentCacheManager cacheManager = createCacheManager();
+    Cache<Long, String> cache = cacheManager.getCache(CACHE_NAME, Long.class, String.class);
+
+    for (int i = 0; i < 10; i++) {
+      cache.put(KEY, String.valueOf(i));
+    }
+
+    assertValue(cache, String.valueOf(9));
+
+    CLUSTER.getClusterControl().terminateActive();
+    CLUSTER.getClusterControl().waitForActive();
+    CLUSTER.getClusterControl().startOneServer();
+
+    assertValue(cache, String.valueOf(9));
+    checkValueFromLoaderWriter(cache, String.valueOf(9));
+
+    cache.clear();
+  }
+
+  @Test
+  public void testWriteBehindMultipleClients() throws Exception {
+    PersistentCacheManager cacheManager1 = createCacheManager();
+    PersistentCacheManager cacheManager2 = createCacheManager();
+    Cache<Long, String> client1 = cacheManager1.getCache(CACHE_NAME, Long.class, String.class);
+    Cache<Long, String> client2 = cacheManager2.getCache(CACHE_NAME, Long.class, String.class);
+
+    client1.put(KEY, "The one from client1");
+    client2.put(KEY, "The one one from client2");
```

Comment:
The one one?

Reply:
I meant here, why are we repeating one twice? The message could be "The one form client2" no?

Reply:
changed now